docs/root/intro/arch_overview/load_balancing/load_balancers.rst

```diff
@@ -56,23 +56,29 @@ weight greater than 1.
 Ring hash
 ^^^^^^^^^
 
-The ring/modulo hash load balancer implements consistent hashing to upstream hosts. The algorithm is
-based on mapping all hosts onto a circle such that the addition or removal of a host from the host
-set changes only affect 1/N requests. This technique is also commonly known as `"ketama"
-<https://github.com/RJ/ketama>`_ hashing. A consistent hashing load balancer is only effective
-when protocol routing is used that specifies a value to hash on. The minimum ring size governs the
-replication factor for each host in the ring. For example, if the minimum ring size is 1024 and
-there are 16 hosts, each host will be replicated 64 times. The ring hash load balancer does not
-currently support weighting.
+The ring/modulo hash load balancer implements consistent hashing to upstream hosts. Each host is
+mapped onto a circle (the "ring") by hashing its address; each request is then routed to a host by
+hashing some property of the request, and finding the nearest corresponding host clockwise around
+the ring. This technique is also commonly known as `"Ketama" <https://github.com/RJ/ketama>`_
+hashing, and like all hash-based load balancers, it is only effective when protocol routing is used
+that specifies a value to hash on.
+
+Each host is hashed and placed on the ring some number of times proportional to its weight. For
+example, if host A has a weight of 1 and host B has a weight of 2, then there might be three entries
+on the ring: one for host A and two for host B. This doesn't actually provide the desired 2:1
+partitioning of the circle, however, since the computed hashes could be coincidentally very close to
+one another; so it is necessary to multiply the number of hashes per host---for example inserting
+100 entries on the ring for host A and 200 entries for host B---to better approximate the desired
+distribution. Best practice is to explicitly set
+:ref:`minimum_ring_size<envoy_api_field_Cluster.RingHashLbConfig.minimum_ring_size>` and
+:ref:`maximum_ring_size<envoy_api_field_Cluster.RingHashLbConfig.maximum_ring_size>`, and monitor
+the min_hashes_per_host and max_hashes_per_host gauges to ensure good distribution. With the ring
```

Comment:
nit: Can you ref link to the stats docs? Edit, I don't think there are stat docs for the new stats so please add those also. :) (Take a look at generally how we do stat docs for examples.)

Reply:
Done, thanks for the pointer.